corehq/apps/hqmedia/views.py

```diff
@@ -186,13 +186,57 @@ def upload_controllers(self):
 
     def get(self, request, *args, **kwargs):
         if request.GET.get('json', None):
-            return JsonResponse({
-                "references": self.app.get_references(),
-                "object_map": self.app.get_object_map(),
-                "totals": self.app.get_reference_totals(),
-            })
+            only_missing = bool(int(request.GET.get('only_missing', 0)))
+            include_total = bool(int(request.GET.get('include_total', 0)))
+            limit = int(request.GET.get('limit', 5))
+            page = int(request.GET.get('page', 1))
+
+            (total, references) = self._process_references(page,
+                                                           limit,
+                                                           only_missing=only_missing,
+                                                           include_total=include_total)
+            multimedia_map = {r['path']: self.app.multimedia_map[r['path']]
+                              for r in references
+                              if r['path'] in self.app.multimedia_map}
+            object_map = self.app.get_object_map(multimedia_map=multimedia_map)
+
+            data = {
+                "references": references,
+                "object_map": object_map,
+            }
+            if include_total:
+                data.update({
+                    "totals": self.app.get_reference_totals(),
+                    "total_rows": total,
+                })
+            return JsonResponse(data)
         return super(MultimediaReferencesView, self).get(request, *args, **kwargs)
 
+    def _process_references(self, page, limit, only_missing=False, include_total=True):
```

Comment:
also I think its confusing that `include_total` defaults to `True` in code, but it defaults to `False` if this isn't specified in the URL

Reply:
Updated in 5145914362c